src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -7533,9 +7533,21 @@ private StatementSyntax ParseEmbeddedStatement(bool complexCheck)
                 case SyntaxKind.LabeledStatement:
                 case SyntaxKind.LocalDeclarationStatement:
                 case SyntaxKind.LocalFunctionStatement:
-                case SyntaxKind.DeconstructionDeclarationStatement:
                     statement = this.AddError(statement, ErrorCode.ERR_BadEmbeddedStmt);
                     break;
+                case SyntaxKind.ExpressionStatement:
+                    // Deconstruction-declaration is only allowed as top-level statement
+                    // see https://github.com/dotnet/roslyn/issues/15049
+                    var expression = ((ExpressionStatementSyntax)statement).Expression;
+                    if (expression.Kind == SyntaxKind.SimpleAssignmentExpression)
+                    {
+                        var assignment = (AssignmentExpressionSyntax)expression;
+                        if (assignment.Left.EnumerateNodes().Any(x => x.RawKind == (int)SyntaxKind.DeclarationExpression))
```

Comment:
Can we simplify the check by simply checking if the left is a tuple or a declaration expression?